### PR TITLE
add scripts to import and run tests from whitequark/parser

### PR DESCRIPTION
**Pull request by iliabylich**
Part of https://github.com/sorbet/sorbet/pull/1670.

I'm going to move two directions in parallel:
1. scripts, tests and infrastructure to run them
2. updates to the grammar

Otherwise I may end up having overlapping PRs.

This PR includes scripts to import and run tests from whitequark/parser.

I haven't generated them yet, going to do it in the next PR.
I hope there are no tools for DCE 😄 , this code is not used atm.

### Test plan

See included automated tests.

### CODE REVIEW

File: test/import_whitequark.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
+
+# This script imports test cases from the whitequark/parser repo.
+# It should be run manually, and it's not run by any automated test suite.
+# When executed on fresh master it should produce zero changse.
```
Comment (jez) [inline test/import_whitequark.sh]:
```suggestion
# When executed on fresh master it should produce zero changes.
```

File: test/import_whitequark.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#
+# To bump version of the whitequark/parser that sorbet/parser mirrors
+# change "REF" variable and re-run the script. Then update the code to match tests.
+#
+# The script runs whitequark/parser tests only for a version of Ruby
+# specified in "TARGET_RUBY_VERSION" variable, bump it if you want to upgrade
+# Sorbet's parser to the next version
```
Comment (jez) [inline test/import_whitequark.sh]: Thanks for this comment!

File: test/import_whitequark.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+# Drop and re-create a directory with tests
+rm -rf "$IMPORT_TO"
+mkdir -p "$IMPORT_TO"
+
+# Run importer
+TARGET_RUBY_VERSION="$TARGET_RUBY_VERSION" \
+    IMPORT_FROM="$IMPORT_FROM" \
+    IMPORT_TO="$IMPORT_TO" \
+    ruby "$ROOT/test/import_whitequark.rb"
```
Comment (jez) [inline test/import_whitequark.sh]: For discoverability, you might want to have this script live in `tools/scripts/`, which is where we keep scripts that are meant to be run interactively (most things in `test/` are run indirectly via bazel).

Up to you whether you want to move it or keep it.

File: test/test_corpus.cc [hunk 4]
```diff
@@ -498,6 +512,117 @@ TEST_P(ExpectationTest, PerPhaseTest) { // NOLINT
     TEST_COUT << "errors OK" << '\n';
 } // namespace sorbet::test
 
+TEST_P(WhitequarkParserTest, PerPhaseTest) { // NOLINT
```
Comment (jez) [inline test/test_corpus.cc]: Do I read this PR right that this section is never tested in CI? I think that our `bazel` runner (`test/BUILD`) only creates test suites for `PosTests` and `LSPTests`.

It's fine if that's the case, I just wanted to make sure that I'm understanding that you didn't mean for these tests to be run yet.

Comment (iliabylich) [reply test/test_corpus.cc]: Yes, #1670 has the following code in `test/BUILD`:
```
pipeline_tests(
    "whitequark_parser_corpus",
    glob([
        "whitequark/**/*.rb",
        "whitequark/**/*.exp",
    ]),
    "WhitequarkParserTests",
    filter = "WhitequarkParserTests",
)
```

I was going to run the script and generate all these tests in a separate PR. Right now this code is not used

File: test/import_whitequark.sh (hunk 3 above)
Comment (iliabylich) [reply test/import_whitequark.sh]: Yep, makes sense, I'll move them, don't run the CI yet please

**Comment by iliabylich** [comment]
@jez Done, ready for merging

**Comment by iliabylich** [comment]
@jez could you merge it please? I'd like to send another PR with generated tests and rebase https://github.com/sorbet/sorbet/pull/1670 to have a small diff

**Comment by jez** [comment]
Thanks!

**Pull request merged**